plugins/modules/v_switch.py

```diff
@@ -0,0 +1,473 @@
+    elif error == 'VSWITCH_IN_PROCESS':
+        module.fail_json(msg='There is a update running, therefore the vswitch can not be updated')
+    elif error == 'VSWITCH_SERVER_LIMIT_REACHED':
+        module.fail_json(msg='The maximum number of servers is reached for this vSwitch')
+
+    # TODO: add and delete with `wait=false`
+    wait_condition = is_all_servers_ready if module.params['wait'] else None
+    return get_v_switch(module, id_, wait_condition)
+
+
+def delete_servers(module, id_, servers):
+    url = '{0}/{1}/server'.format(V_SWITCH_BASE_URL, id_)
+    headers = {'Content-type': 'application/x-www-form-urlencoded'}
+    data = get_x_www_form_urlenconded_dict_from_list('server', servers)
+    result, error = fetch_url_json(
+        module,
+        url,
+        data=urlencode(data),
+        headers=headers,
+        method='DELETE',
+        # TODO: missing INVALID_INPUT, NOT_FOUND
+        accept_errors=['SERVER_NOT_FOUND', 'VSWITCH_IN_PROCESS'],
+        allow_empty_result=True,
+    )
+    if error == 'SERVER_NOT_FOUND':
+        # information about which servers are failing is only there
+        module.fail_json(msg=result['error']['message'])
+    elif error == 'VSWITCH_IN_PROCESS':
+        module.fail_json(msg='There is a update running, therefore the vswitch can not be updated')
+
+    wait_condition = is_all_servers_ready if module.params['wait'] else None
+    return get_v_switch(module, id_, wait_condition)
+
+
+def get_servers_to_delete(current_servers, desired_servers):
+    return [
+        server['server_ip']
+        for server in current_servers
+        if server['server_ip'] not in desired_servers
+        and str(server['server_number']) not in desired_servers
+    ]
+
+
+def get_servers_to_add(current_servers, desired_servers):
+    current_ids = [str(server['server_number']) for server in current_servers]
+    current_ips = [server['server_ip'] for server in current_servers]
+    return [
+        server
+        for server in desired_servers
+        if server not in current_ips and server not in current_ids
+    ]
+
+
+def set_desired_servers(module, id_):
+    v_switch = get_v_switch(module, id_)
+    changed = False
+
+    servers_to_delete = get_servers_to_delete(v_switch['server'], module.params['servers'])
+    if len(servers_to_delete) > 0:
+        v_switch = delete_servers(module, id_, servers_to_delete)
+        changed = True
+    servers_to_add = get_servers_to_add(v_switch['server'], module.params['servers'])
+    if len(servers_to_add) > 0:
+        v_switch = add_servers(module, id_, servers_to_add)
+        changed = True
+    return (v_switch, changed)
+
+
+def main():
+    argument_spec = dict(
+        vlan=dict(type='int', required=True),
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        servers=dict(type='list', elements='str', default=[]),
```

Comment:
Shouldn't the default for `servers` be `None` (i.e. not specified), so that if the user just says "state: present" the module won't clear the server list for that vSwitch?

Reply:
I tried changing it to None and adding that logic but it seems that then this happens in sanity tests:
```
ERROR: plugins/modules/v_switch.py:0:0: doc-default-does-not-match-spec: Argument 'servers' in argument_spec defines default as (None) but documentation defines default as (['None'])
```
I was checking other list default configurations and they seems to be `[]`.

Reply:
I didn't think in removing the default value. Thanks for your advice. Fixed in 54cf6c6ea626ccd0f2b332e1c644e068a710f377.